modules/playlists/src/main/java/org/opencastproject/playlists/PlaylistRestService.java

```diff
@@ -0,0 +1,771 @@
+  protected SearchService searchService = null;
+
+  /** The authorization service */
+  protected AuthorizationService authorizationService = null;
+
+  /**
+   * Sets the playlist service
+   *
+   * @param service
+   *          the playlist service instance
+   */
+  @Reference
+  public void setService(PlaylistService service) {
+    this.service = service;
+  }
+
+  @Reference
+  protected void setSearchService(SearchService searchService) {
+    this.searchService = searchService;
+  }
+
+  @Reference
+  public void setAuthorizationService(AuthorizationService authorizationService) {
+    this.authorizationService = authorizationService;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("{id}.json")
+  @RestQuery(
+      name = "playlist",
+      description = "Get a playlist.",
+      returnDescription = "A playlist as JSON",
+      pathParameters = {
+          @RestParameter(name = "id", isRequired = true, description = "The playlist identifier", type = STRING),
+      },
+      restParameters = {
+          @RestParameter(name = "withPublications", isRequired = false, description = "If available publications for"
+              + "the content should be returned. Only works for content of type EVENT.", type = BOOLEAN,
+              defaultValue = "true")
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "A playlist as JSON."),
+          @RestResponse(responseCode = SC_NOT_FOUND, description = "No playlist with that identifier exists."),
+          @RestResponse(responseCode = SC_UNAUTHORIZED, description = "Not authorized to perform this action")
+      })
+  public Response getPlaylistAsJson(
+      @PathParam("id") String id,
+      @FormParam("withPublications") boolean withPublications)
+          throws NotFoundException, UnauthorizedException {
+    Playlist playlist = service.getPlaylistById(id);
+
+    JaxbPlaylist jaxbPlaylist;
+    if (withPublications) {
+      jaxbPlaylist = service.enrich(playlist);
+    } else {
+      jaxbPlaylist = new JaxbPlaylist(playlist);
+    }
+
+    return Response.ok().entity(jaxbPlaylist).build();
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_XML)
+  @Path("{id}.xml")
+  @RestQuery(
+      name = "playlist",
+      description = "Get a playlist.",
+      returnDescription = "A playlist as XML",
+      pathParameters = {
+          @RestParameter(name = "id", isRequired = true, description = "The playlist identifier", type = STRING),
+      },
+      restParameters = {
+          @RestParameter(
+              name = "withPublications",
+              isRequired = false,
+              description = "If available publications for"
+              + "the content should be returned. Only works for content of type EVENT.",
+              type = BOOLEAN,
+              defaultValue = "true"
+          )
+      },
+      responses = {
+          @RestResponse(responseCode = SC_OK, description = "A playlist as XML."),
```

Comment:
I think this method should return the status code [201 Created](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201).